plugins/jobs/web_client/views/JobListWidget.js

```diff
@@ -90,99 +157,246 @@ var JobListWidget = View.extend({
     ], 'COLUMN_ALL'),
 
     render: function () {
-        var jobs, types, states;
-
-        types = _.uniq(this.collection.toArray().map(function (job) {
-            return job.attributes.type ? job.attributes.type : '';
-        }));
-
-        this._updateFilter(this.typeFilter, types);
-        this.filterTypeMenuWidget.setValues(this.typeFilter);
-
-        states = _.uniq(this.collection.toArray().map(function (job) {
-            return JobStatus.text(job.attributes.status);
-        }));
-        this._updateFilter(this.statusFilter, states);
-        this.filterStatusMenuWidget.setValues(this.statusFilter);
-
-        jobs = this._filterJobs(this.collection.toArray());
-
-        this.$el.html(JobListWidgetTemplate({
-            jobs: jobs,
-            showHeader: this.showHeader,
-            columns: this.columns,
-            columnEnum: this.columnEnum,
-            linkToJob: this.linkToJob,
-            triggerJobClick: this.triggerJobClick,
-            JobStatus: JobStatus,
-            formatDate: formatDate,
-            DATE_SECOND: DATE_SECOND
-        }));
-
-        this.filterTypeMenuWidget.setElement(this.$('.g-job-type-header')).render();
-        this.filterStatusMenuWidget.setElement(this.$('.g-job-status-header')).render();
+        this.$el.html(JobListTemplate(this));
+
+        this.typeFilterWidget.setElement(this.$('.filter-container .type')).render();
+        this.statusFilterWidget.setElement(this.$('.filter-container .status')).render();
+
+        this.$('a[data-toggle="tab"]').on('shown.bs.tab', e => {
+            this.currentView = $(e.target).attr('name');
+            if (this.userId) {
+                router.navigate(`jobs/user/${this.userId}/${this.currentView}`);
```

Comment:
It's nothing to do with performance and everything to do with maintainability/readability imo. We have a view that has different "types" of views within it, that's not very intuitive to the next person who will work on this code. For example, to change something on the list view the developer now has to wade through the logic of multiple graphs. If this were a view that created child views which housed that logic, that would be more intuitive.